pulp_container/app/authorization.py

```diff
@@ -133,6 +133,13 @@ def determine_access(self):
         actions = set(actions.split(","))
 
         permitted_actions = set()
+        if "push" in actions:
+            has_permission = self.actions_permissions["push"](name)
+            if has_permission:
+                permitted_actions.add("push")
+                permitted_actions.add("pull")
+
+            return {"type": typ, "name": name, "actions": list(permitted_actions)}
```

Comment:
What's the reason for this codepath? In my experience podman will ask for push and pull which will be handled in the loop after this.

Reply:
imagine you are pushing something for the first time. You don't have the distribution created just yet. This code path is needed to skip has_pull_permission which will fail unless the user has a global 'pull' permission..which does not really sound logical to give to a user ahead of push the global pull permission. This actually was the attempt to fix the but where in order to push we neded to give add_namespace, add_distro and view_distro perms https://github.com/pulp/pulp_container/pull/203/files#diff-18f14684e2e23997ce0169f5a491f334fee88119dfbe8587594359fe72c10bc0R40
In reality it should be enough to give add_namespace and  add_distro only